compiler/rustc_hir_analysis/src/check/check.rs

```diff
@@ -781,6 +781,11 @@ fn check_impl_items_against_trait<'tcx>(
             trait_def.must_implement_one_of.as_deref();
 
         for &trait_item_id in tcx.associated_item_def_ids(impl_trait_ref.def_id) {
+            // Do not consider synthesized RPITITs on missing items checks.
+            if tcx.opt_rpitit_info(trait_item_id).is_some() {
+                continue;
+            }
```

Comment:
Can we just filter the RPITITs in `missing_items_err` instead? Then we will always make sure that we're emitting an error.

Reply:
My philosophy is that I'd rather move the RPITIT-specific logic as close to the diagnostic code as possible, *just* in case we have some bug like we forget to synthesize an impl associated type even when the corresponding fn is present :thinking: 

I know that's a really small possibility, but both changes seem equally short.

Reply:
There are two parts here one is the error missing `` implementation and the other part is the `` from trait. Going to push what's needed to fix both things by filtering for you to consider what's better.

Reply:
@compiler-errors, check it out if this version is better or if we should stick with https://github.com/rust-lang/rust/pull/109414/commits/b539028ea11642303dc79ae45c483a7490b8fa4e. But I do agree with you that the new one is better.